android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java

```diff
@@ -590,6 +632,37 @@ public void didSelectTab(Tab tab, @TabSelectionType int type, int lastId) {
                 };
     }
 
+    private void showYouTubePipIcon(@NonNull final Tab tab) {
+        // The layout could be null in Custom Tabs layout.
+        if (mYouTubePipLayout == null) {
+            return;
+        }
+
+        // Hide the layout if the current tab is in a state where it doesn't support or allow PiP
+        // mode. This can also happen when a tab is re-selected after a crash and it's showing
+        // the crash custom view, or is in a frozen state (likely inactive or unloaded).
+        if (!BraveYouTubeScriptInjectorNativeHelper.isPictureInPictureAvailable(
+                tab.getWebContents())) {
+            mYouTubePipLayout.setVisibility(View.GONE);
+            return;
+        }
+        final boolean show =
+                PictureInPicture.isEnabled(getContext())
+                        && ChromeFeatureList.isEnabled(
```

Comment:
this check should move inside `isPictureInPictureAvailable`

Reply:
Addressed  by commit 👉 [`2068f4a` (#29431)](https://github.com/brave/brave-core/pull/29431/commits/2068f4a83b143d20895b3a2818fde9e7d8b5628d)